libsolidity/interface/AssemblyStack.h

```diff
@@ -0,0 +1,83 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * Full assembly stack that can support EVM-assembly and JULIA as input and EVM, EVM1.5 and
+ * eWasm as output.
+ */
+
+#pragma once
+
+#include <libsolidity/interface/Exceptions.h>
+#include <libsolidity/inlineasm/AsmAnalysisInfo.h>
+#include <libevmasm/LinkerObject.h>
+
+#include <string>
+#include <memory>
+
+namespace dev
+{
+namespace solidity
+{
+class Scanner;
+namespace assembly
+{
+struct AsmAnalysisInfo;
+struct Block;
+}
+
+/*
+ * Full assembly stack that can support EVM-assembly and JULIA as input and EVM, EVM1.5 and
+ * eWasm as output.
+ */
+class AssemblyStack
+{
+public:
+	enum class Language { JULIA, Assembly };
+	enum class Machine { EVM, EVM15, eWasm };
+
+	explicit AssemblyStack(Language _language = Language::Assembly):
+		m_language(_language)
+	{}
+
+	/// @returns the scanner used during parsing
+	Scanner const& scanner() const;
+
+	/// Runs parsing and analysis steps, returns false if input cannot be assembled.
```

Comment:
Add note: "Subsequent calls overwrite the previous state".

Alternatively we could add a `reset()` and asserts in `parseAndAnalyze` to be the empty state?